lib/rouge/lexers/email.rb

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Email < RegexLexer
+      tag 'email'
+      aliases 'eml', 'e-mail'
+      filenames '*.eml'
+      mimetypes 'message/rfc822'
+
+      title "Email"
+      desc "An email message"
+
+      start do
+        push :fields
+      end
+
+      state :fields do
+        rule %r/[:]/, Operator, :field_body
+        rule %r/[^\n\r:]+/, Name::Tag
+        rule %r/[\n\r]/, Name::Tag
+      end
+
+      state :field_body do
+        rule(/(\r?\n){2}/) { token Text; pop!(2) }
+        rule %r/\r?\n(?![ \v\t\f])/, Text, :pop!
+        rule %r/[^\n\r]+/, Name::Attribute
+        rule %r/[\n\r]/, Name::Attribute
+      end
+
+      state :root do
+        rule %r/.*/m, Text
```

Comment:
@smokris You were lexing quoted passages and signatures differently from the other text in the body of the e-mail. I realise that is useful for your use case but I don't think it's appropriate to make that part of the lexer given it's not part of the spec.

Reply:
Trying to find quotes and reply parts and signatures... that way madness lies. It's locale-dependent, and can be literally any characters ><

https://github.com/discourse/email_reply_trimmer/blob/master/lib/email_reply_trimmer/embedded_email_matcher.rb#L13

Reply:
@pyrmont and @jneen, thanks for your work on this!

> [jneen] https://github.com/discourse/email_reply_trimmer/blob/master/lib/email_reply_trimmer/embedded_email_matcher.rb#L13

Yikes.  Yeah, I don't think Rouge should attempt that level of… um… sophistication.

For my use-case, I'm less concerned about signatures and "on date someone wrote" lines, but we do frequently use inline-reply quoted text (lines beginning with greater-than, like in `spec/visual/samples/email`), and I think it significantly improves readability to highlight that quoted text differently from unquoted text.

> [pyrmont] I don't think it's appropriate to make that part of the lexer given it's not part of the spec.

It's not in RFC (2)822, but [RFC 3676 section 4.5](https://tools.ietf.org/html/rfc3676#section-4.5) does specify this:

> the canonical quote indicator (or quote mark) is one or more close angle bracket (">") characters.  Lines which start with the quote indicator are considered quoted.  The number of ">" characters at the start of the line specifies the quote depth.

I think that much is unambiguous, so I'm still in favor of highlighting (well, "dimming") quoted text.

Reply:
@smokris Sorry for the delay in responding. I've added support for quoted lines on the basis of the section in RFC 3676 that you linked.